### ISSUE TITLE
[FEATURE] Allow Harvester Node Promotion from GUI or Kubectl

### ISSUE THREAD

**Issue by dipakparmar** [harvester/harvester#5421]
**Is your feature request related to a problem? Please describe.**
Harvester version 1.2.1 automatically promotes nodes to management and compute roles, but in a DR scenario, this promotion does not happen automatically, causing cluster breakage even with VIP setup.

**Describe the solution you'd like**
Implement automatic node promotion to management and compute roles in DR scenarios to prevent cluster breakdown and ensure seamless failover. Also, documentation on how to manually do the node promotion and button from GUI.

**Describe alternatives you've considered**
Tried removing  node-role annotations and adding them back, as well as managing HCI node roles, but encountered issues like status staying in restarting and cluster breakage.

**Additional context**
The current manual workaround of restoring backup nodes is cumbersome and disrupts cluster operations. Automating node promotion in DR scenarios will enhance cluster resilience and simplify recovery processes.

**Comment by notsrch** [comment]
This is very important for us.

We had a master node hardware failure (non-recoverable) on a master node.   While troubleshooting, a second master node went down.

**Comment by bk201** [comment]
@Vicente-Cheng I believe we can specify the role now, but not via GUI.

**Comment by Vicente-Cheng** [comment]
Yes, we introduce the role management on v1.3.0. Reference as below:
https://docs.harvesterhci.io/v1.3/install/harvester-configuration#installrole
https://docs.harvesterhci.io/v1.3/host/#role-management

Hi @dipakparmar, @notsrch
Could you check the document above and give some feedback?
Or would you like to share more about your scenario?

Thanks!

**Comment by notsrch** [comment]
@Vicente-Cheng 
We had an 8 node harvester cluster running v1.3.1.   Nodes 1-3 were the master nodes.   Node#2 had a hardware failure.   3 days later, while troubleshooting Node#2, Node#3 failed.   Both were hardware failures and unrecoverable.

At least in my mind, a failure of only 2 nodes should not lead to a down and non-recoverable cluster.   It should be possible to promote a new node to master without deleting the failed one.   Ideally it should also happen automatically.

**Comment by kfehrenbach** [comment]
Because of split brain scenarios i would suggest not to promote a node to master automatically. But there needs to be a solution to do it manually!

**Comment by notsrch** [comment]
Another option that would also help for our use case is having 5 master nodes instead of 3.   Anything that helps survive through a double hardware failure or allow for quicker recovery from it is a net gain.

**Comment by w13915984028** [comment]
The label `disaster-recovery` (DR) is added.

Both https://github.com/harvester/harvester/issues/5421 and https://github.com/harvester/harvester/issues/5424 will be taken into account for further enhancements:
(1) Allow user to set e.g. 5 management nodes; the node-promoter can work for more nodes promotion. Also allow to set mode like `auto-promotion` and `manual-promotion`.
(2) Change the node-role on the fly.
(3) Different number of management role and etcd role, e.g. 3 mgmt & 5 etcd, or 5 mgmt & 3 etcd.

TBD...

Any feedbacks are welcome.

**Comment by Vicente-Cheng** [comment]
still relevant

**Comment by forbesguthrie** [comment]
I think this is a reasonable request and comes up quite frequently with customers. I assume from previous comments that this adding options to the UI is the only part that's left to do, in which case we should scope this out and add this to the backlog. Let me know if it's more complex than this.

**Comment by hoo29** [comment]
We recently had a situation where harvester failed to promote a node. @ParadoxGuitarist wrote a great guide on https://rancher-users.slack.com/archives/C01GKHKAG0K/p1777046166753689?thread_ts=1776811103.034029&cid=C01GKHKAG0K that worked perfectly for us. It would be good to get that added.

**Comment by ParadoxGuitarist** [comment]
Because slack has a tendency to disappear messages after a few months, here's the tldr section from the bottom of that slack thread: 

- Get the machine ID for the node you're trying to promote (gpu1 in this example) : `MACHINE_ID=$(kubectl get node gpu1 -o jsonpath='{.metadata.annotations.cluster\.x-k8s\.io/machine}') ; echo $MACHINE_ID`
- There's some files you need to create on that node. Copy over them from an existing good node: `/etc/rancher/rke2/config.yaml.d/90-harvester-server.yaml` and `/etc/rancher/rke2/config.yaml.d/40-disable-charts.yaml`
- Label the Machine object `kubectl label --overwrite -n fleet-local machines.cluster.x-k8s.io $MACHINE_ID   rke.cattle.io/control-plane-role=true   rke.cattle.io/etcd-role=true   cluster.x-k8s.io/control-plane=true`
- Label the Machine Plan Secret `kubectl label --overwrite -n fleet-local secret ${MACHINE_ID}-machine-plan  rke.cattle.io/control-plane-role=true rke.cattle.io/etcd-role=true`
- Label the RKE Bootstrap object `kubectl label --overwrite -n fleet-local rkebootstraps.rke.cattle.io $MACHINE_ID rke.cattle.io/control-plane-role=true rke.cattle.io/etcd-role=true`
- Wait a few minutes and then add the master role to the node and verify they all look the same: `kubectl label node gpu1 node-role.kubernetes.io/master=true ; kubectl get nodes`
